api/lib/round-tracker.js

```diff
@@ -245,10 +252,30 @@ export async function maybeCreateSparkRound (pgClient, {
     meridianContractAddress,
     meridianRoundIndex,
     roundStartEpoch,
-    MAX_TASKS_PER_NODE
+    BASELINE_MAX_TASKS_PER_NODE
   ])
 
   if (rowCount) {
+    if (useDynamicTaskCount) {
+      try {
+        const res = await fetch('https://stats.filspark.com/stations/daily')
+        const [{ station_id_count: stationIdCount }] = await res.json()
+        if (stationIdCount === 0) {
+          throw new Error('No active stations found')
+        }
+        await pgClient.query(`
+          UPDATE spark_rounds
+          SET max_tasks_per_node = $1
+          WHERE id = $2  
```

Comment:
I don't like this two-step process where you update the value after it was written to the DB. There is a race condition - if a spark checker requests the current rounds data after you inserted the row but before your updated `max_tasks_per_node`, Cloudflare will store the incorrect response in the cache and serve that incorrect response to other clients (checker nodes) running in the same region.

The fact that there is an HTTP request between these two SQL queries makes that window even wider.

Let's move the code calculating `max_tasks_per_node` to the top, so that the INSERT statement can set the correct `max_task_per_node` value.